tensorboard/plugins/mesh_visualizer/mesh_plugin.py

```diff
@@ -0,0 +1,253 @@
+        self._tag_to_instance_tags[meta.name].append(tag)
+        self._instance_tag_to_tag[tag] = meta.name
+
+  @wrappers.Request.application
+  def _serve_tags(self, request):
+    """A route (HTTP handler) that returns a response with tags.
+
+    Args:
+      request: The werkzeug.Request object.
+
+    Returns:
+      A response that contains a JSON object. The keys of the object
+      are all the runs. Each run is mapped to a (potentially empty)
+      list of all tags that are relevant to this plugin.
+    """
+    # This is a dictionary mapping from run to (tag to string content).
+    # To be clear, the values of the dictionary are dictionaries.
+    all_runs = self._multiplexer.PluginRunToTagToContent(
+        MeshPlugin.plugin_name)
+
+    # Make sure we populate tags mapping structures.
+    self.prepare_metadata()
+
+    # tagToContent is itself a dictionary mapping tag name to string
+    # SummaryMetadata.plugin_data.content. Retrieve the keys of that dictionary
+    # to obtain a list of tags associated with each run. For each tag estimate
+    # number of samples.
+    response = dict()
+    for run, tag_to_content in six.iteritems(all_runs):
+      response[run] = dict()
+      for instance_tag, _ in six.iteritems(tag_to_content):
+        # Make sure we only operate on user-defined tags here.
+        tag = self._instance_tag_to_tag[instance_tag]
+        meta = self._instance_tag_to_metadata[instance_tag]
+        # Shape should be at least BxNx3 where B represents the batch dimensions
+        # and N - the number of points, each with x,y,z coordinates.
+        assert len(meta.shape) >= 3
+        response[run][tag] = {'samples': meta.shape[0]}
+    return http_util.Respond(request, response, 'application/json')
+
+  def get_plugin_apps(self):
+    """Gets all routes offered by the plugin.
+
+    This method is called by TensorBoard when retrieving all the
+    routes offered by the plugin.
+    Returns:
+      A dictionary mapping URL path to route that handles it.
+    """
+    # Note that the methods handling routes are decorated with
+    # @wrappers.Request.application.
+    return {
+        '/tags': self._serve_tags,
+        '/meshes': self._serve_mesh_metadata,
+        '/data': self._serve_mesh_data
+    }
+
+  def is_active(self):
+    """Determines whether this plugin is active.
+
+    This plugin is only active if TensorBoard sampled any summaries
+    relevant to the mesh plugin.
+    Returns:
+      Whether this plugin is active.
+    """
+    all_runs = self._multiplexer.PluginRunToTagToContent(
+        MeshPlugin.plugin_name)
+
+    # The plugin is active if any of the runs has a tag relevant
+    # to the plugin.
+    return bool(self._multiplexer and any(six.itervalues(all_runs)))
+
+  def _get_sample(self, tensor_event, sample):
+    """Returns a single sample from a batch of samples."""
+    data = tf.make_ndarray(tensor_event.tensor_proto)
```

Comment:
Please replace `tf.make_ndarray` to TensorBoard one and remove the TensorFlow dependency. 
https://github.com/tensorflow/tensorboard/blob/0da00e7d1cd8e171402a848cc33f0554a25a0628/tensorboard/plugins/audio/audio_plugin.py#L30 

Motivation: we are trying to break TensorFlow dependency as much as possible